pkg/routeagent/controllers/route/route.go

```diff
@@ -24,30 +32,74 @@ type Controller struct {
 	clusterID       string
 	objectNamespace string
 
-	submarinerClientSet clientset.Interface
-	clustersSynced      cache.InformerSynced
-	endpointsSynced     cache.InformerSynced
+	submarinerClientSet    clientset.Interface
+	clientSet              *kubernetes.Clientset
+	clustersSynced         cache.InformerSynced
+	endpointsSynced        cache.InformerSynced
+	smRouteAgentPodsSynced cache.InformerSynced
 
 	clusterWorkqueue  workqueue.RateLimitingInterface
 	endpointWorkqueue workqueue.RateLimitingInterface
+	podWorkqueue      workqueue.RateLimitingInterface
 
-	gw      net.IP
-	subnets []string
+	gatewayNodeIP    net.IP
+	localClusterCidr []string
+	localServiceCidr []string
+	remoteSubnets    *util.StringSet
```

Comment:
`remoteSubnets` is shared between the _Cluster_ and _Endpoint_  processing.  `StringSet` is thread-safe so concurrent access isn't a problem. However `remoteSubnets` is populated via the _Cluster_  processing so will this cause an issue with ordering, eg if an _Endpoint_ is observed before a _Cluster_  and `remoteSubnets` is not yet populated?